plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -17,149 +18,55 @@
 
-# pylint: disable=too-many-branches
-def make_nevergrad_parameter(description: Any) -> Any:
-    """Returns a Nevergrad parameter from a definition string or object.
+    # ranges are always ordered:
+    range(1,12)  =>  ng.p.Scalar(lower=1, upper=12, step=1)
 
-    Parameters
-    ----------
-    description: Any
-       * a commandline definition string. This can be:
-         - comma-separated values: for a choice parameter
-           Eg.: "a,b,c"
-           Note: sequences of increasing scalars provide a specific parametrization
-             compared to unordered categorical values
-         - ":"-separated values for ranges of scalars.
-           "int" and/or "log" modifiers can be added in front to cast to integer or
-           use log-distributed values (Eg: int:log:4:1024)
-         - anything else will be treated as a constant string
-       * a config definition dict for scalar parameters, with potential fields
-         init, lower, upper, step, log, integer
-       * a list for option parameters defined in config file
+    # intervals are scalars:
+    # Note: by intervals are always interpreted as floats (even for int start and end values).
+    interval(0,1)     -> RangeSweep(start=0.0, end=1.0) -> ng.p.Scalar(lower=0.0, upper=1.0)
+    interval(0.0,1.0) -> RangeSweep(start=0.0, end=1.0) -> ng.p.Scalar(lower=0.0, upper=1.0)
 
-    Returns
-    -------
-    Parameter or str
-        A Parameter if the string fitted one of the definitions, else the input string
+    # a user can cast the interval to int to override that:
+    int(interval(0,1) -> RangeSweep(start=0, end=1) -> ng.p.Scalar(lower=0.0, upper=1.0).set_integer_casting()
     """
-    # lazy initialization to avoid overhead when loading hydra
-    import nevergrad as ng
-
-    # revert config parsing
-
-    if isinstance(description, (ListConfig, list)):
-        description = ",".join(str(x) for x in description)
-    if isinstance(description, str):
-        # cast to spec if possible
-        try:
-            description = CommandlineSpec.parse(description)
-        except ValueError:
-            pass
-    # convert scalar commandline specs to dict
-    if isinstance(description, CommandlineSpec) and description.bounds is not None:
-        description = ScalarConfigSpec(
-            lower=description.bounds[0],
-            upper=description.bounds[1],
-            log=description.log,
-            integer=description.cast == "int",
-        )
-    # convert scalar config specs to dict
-    # convert dict to Scalar parameter instance
+    scalar = None
+    if isinstance(description, Override):
+        override = description
+        val = override.value()
+        if override.is_sweep_override():
+            if override.is_choice_sweep():
+                if "ordered" in val.tags:
+                    return ng.p.TransitionChoice(val.list)
+                else:
+                    return ng.p.Choice(val.list)
```

Comment:
1. What happens on shuffle() and sort()?
2. What happens if the values are lists or dicts? (`[1,2],[3,4]`, `{foo:10},{bar:20}`)

Follow #945, I think it will solve these problems.

Reply:
For `shuffle` the Sweep class actually have a field indicating the status `RangeSweep(tags=set(), start=1, stop=10, step=1, shuffle=True)`, but for `sort` there's no corresponding field, should we add one for `sort` as well?